projects/igniteui-angular/src/lib/month-picker/month-picker.component.ts

```diff
@@ -0,0 +1,126 @@
+import {
+    Component,
+    NgModule,
+    HostListener
+} from '@angular/core';
+import { IgxCalendarModule, CalendarView, IgxCalendarComponent } from '../calendar/index';
+import { IgxIconModule } from '../icon/index';
+import { NG_VALUE_ACCESSOR } from '@angular/forms';
+import { CommonModule } from '@angular/common';
+import { trigger, transition, useAnimation } from '@angular/animations';
+import { fadeIn, scaleInCenter, slideInLeft, slideInRight } from '../animations/main';
+
+@Component({
+    providers: [
+        {
+            multi: true,
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: IgxMonthPickerComponent
+        }
+    ],
+    animations: [
+        trigger('animateView', [
+            transition('void => 0', useAnimation(fadeIn)),
+            transition('void => *', useAnimation(scaleInCenter, {
+                params: {
+                    duration: '.2s',
+                    fromScale: .9
+                }
+            }))
+        ]),
+        trigger('animateChange', [
+            transition('* => prev', useAnimation(slideInLeft, {
+                params: {
+                    fromPosition: 'translateX(-30%)'
+                }
+            })),
+            transition('* => next', useAnimation(slideInRight, {
+                params: {
+                    fromPosition: 'translateX(30%)'
+                }
+            }))
+        ])
+    ],
+    selector: 'igx-month-picker',
+    templateUrl: 'month-picker.component.html'
+})
+export class IgxMonthPickerComponent extends IgxCalendarComponent {
```

Comment:
Shouldn't we have some public API for this component? An input to set the initial month and an event to notify on selection changes. Etc.

Reply:
We need to import what's exported by this file in the public_api.ts similar to what's done with the calendar component.